### PR TITLE
Np 3495 top level institutions

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
+import static no.unit.nva.cristin.model.Constants.TOP;
+import static no.unit.nva.cristin.model.JsonPropertyNames.DEPTH;
+import static no.unit.nva.cristin.model.JsonPropertyNames.NUMBER_OF_RESULTS;
+import static no.unit.nva.cristin.model.JsonPropertyNames.PAGE;
+import static no.unit.nva.cristin.model.JsonPropertyNames.QUERY;
+
 public abstract class CristinQueryHandler<I, O> extends CristinHandler<I, O> {
 
-    private static final Set<String> VALID_QUERY_PARAMS = Set.of(QUERY, PAGE, NUMBER_OF_RESULTS);
+    private static final Set<String> VALID_QUERY_PARAMS = Set.of(QUERY, PAGE, NUMBER_OF_RESULTS, DEPTH);
```
Comment (torarnet) [inline cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java:24]: Is this a valid query param for all handlers who extend `CristinQueryHandler` ? Maybe you should override it instead?

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java [hunk 2]
```diff
@@ -37,22 +39,32 @@ protected String getValidPage(RequestInfo requestInfo) throws BadRequestExceptio
...
 
     protected String getValidQuery(RequestInfo requestInfo) throws BadRequestException {
         return getQueryParam(requestInfo, QUERY)
             .filter(this::isValidQuery)
             .map(UriUtils::escapeWhiteSpace)
-            .orElseThrow(() -> new BadRequestException(ERROR_MESSAGE_QUERY_MISSING_OR_HAS_ILLEGAL_CHARACTERS));
+            .orElseThrow(() -> new BadRequestException(ErrorMessages.ERROR_MESSAGE_QUERY_MISSING_OR_HAS_ILLEGAL_CHARACTERS));
+    }
+
+    protected String getValidDepth(RequestInfo requestInfo) throws BadRequestException {
```
Comment (torarnet) [inline cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java:60]: Is this gonna be shared or is it specific for Organization query?

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/CristinOrganizationApiClient.java [hunk 3]
```diff
@@ -86,19 +89,24 @@ public Organization getOrganization(URI uri)
             throws NotFoundException, FailedHttpRequestException {
 
         URI queryUri = createCristinQueryUri(translateToCristinApi(requestQueryParams), UNITS_PATH);
-        final long start = System.currentTimeMillis();
         SearchResponse<Organization> searchResponse = query(queryUri);
+        final long start = System.currentTimeMillis();
```
Comment (torarnet) [inline cristin-organization/src/main/java/no/unit/nva/cristin/organization/CristinOrganizationApiClient.java:93]: Why did you move this?

**Review by torarnet — COMMENTED** [review]
See comments

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java (hunk 1 above)
Comment (svenngjeto) [reply cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java]: Query organization code moved to QueryCristinOrganizationHandler

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java (hunk 2 above)
Comment (svenngjeto) [reply cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java]: Same as above

File: cristin-organization/src/main/java/no/unit/nva/cristin/organization/CristinOrganizationApiClient.java (hunk 3 above)
Comment (svenngjeto) [reply cristin-organization/src/main/java/no/unit/nva/cristin/organization/CristinOrganizationApiClient.java]: A result of fat fingers and being asleep while undoing a refactor...

**Review by torarnet — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]
